### DISCUSSION TITLE
Cups server remote driverless setup IOT

### DISCUSSION THREAD

**Discussion by milkman0007** [michaelrsweet/lprint#154] [Q&A]
I was reading about the move to go driverless from OpenPrinting, and it led me down a rabbit hole that I couldn't quite figure out. Companies like PaperCut, PrinterLogic, and others are marketing a driverless environment. They are installing a local driver win or mac with a redirect that sends the data stream to their system for processing. Then, they convert the data into a format that the customer's printer will understand natively and sends it back to the local drivers print quote for the local printer to print. 

My question is: Are they just hosting a CUPS server on the cloud, passing the data via API, and then sending it back to the local printer, or is it something more complex? How are they able to convert formats? I don't think they are writing a profile for every printer out there. Does anyone have any ideas?

**Comment by michaelrsweet** [comment]
That's not something we can answer definitively. Certainly they could be using CUPS under the covers but there are other ways to support printers, too. WRT "writing a profile for every printer out there", that is essentially what we've been doing with CUPS for years.
